app/scripts/controllers/permissions/background-api.test.js

```diff
@@ -1,197 +1,620 @@
+    it('throws an error if there is no existing CAIP-25 caveat', () => {
+      const permissionController = {
+        getCaveat: jest.fn(),
+      };
 
-      expect(permissionController.updateCaveat).toHaveBeenCalledTimes(1);
-      expect(permissionController.updateCaveat).toHaveBeenCalledWith(
-        'foo.com',
-        RestrictedMethods.eth_accounts,
-        CaveatTypes.restrictReturnedAccounts,
-        ['0x1'],
+      expect(() =>
+        getPermissionBackgroundApiMethods({
+          permissionController,
+        }).removePermittedAccount('foo.com', '0x1'),
+      ).toThrow(
+        new Error('tried to remove accounts when none have been permissioned'),
       );
     });
 
-    it('revokes the accounts permission if the removed account is the only permitted account', () => {
+    it('does nothing if the account being removed does not exist', () => {
       const permissionController = {
-        getCaveat: jest.fn().mockImplementationOnce(() => {
-          return {
-            type: CaveatTypes.restrictReturnedAccounts,
-            value: ['0x1'],
-          };
+        getCaveat: jest.fn().mockReturnValue({
+          value: {
+            requiredScopes: {
+              'eip155:1': {
+                methods: [],
+                notifications: [],
+              },
+              'eip155:10': {
+                methods: [],
+                notifications: [],
+                accounts: ['eip155:10:0x1', 'eip155:10:0x2'],
+              },
+            },
+            optionalScopes: {
+              'bip122:000000000019d6689c085ae165831e93': {
+                methods: [],
+                notifications: [],
+                accounts: [
+                  'bip122:000000000019d6689c085ae165831e93:128Lkh3S7CkDTBZ8W7BbpsN3YYizJMp8p6',
+                ],
+              },
+              'eip155:1': {
+                methods: [],
+                notifications: [],
+                accounts: ['eip155:1:0x2', 'eip155:1:0x3'],
+              },
+            },
+            isMultichainOrigin: true,
+          },
         }),
-        revokePermission: jest.fn(),
         updateCaveat: jest.fn(),
+        revokePermission: jest.fn(),
       };
 
-      getPermissionBackgroundApiMethods(
+      getPermissionBackgroundApiMethods({
         permissionController,
-      ).removePermittedAccount('foo.com', '0x1');
+      }).removePermittedAccount('foo.com', '0xdeadbeef');
 
-      expect(permissionController.getCaveat).toHaveBeenCalledTimes(1);
-      expect(permissionController.getCaveat).toHaveBeenCalledWith(
-        'foo.com',
-        RestrictedMethods.eth_accounts,
-        CaveatTypes.restrictReturnedAccounts,
-      );
+      expect(permissionController.updateCaveat).not.toHaveBeenCalled();
+      expect(permissionController.revokePermission).not.toHaveBeenCalled();
+    });
+
+    it('revokes the permission if the removed account is the only eth account', () => {
```

Comment:
🤔 I suppose this okay for now.
```suggestion
    it('revokes the entire permission if the removed account is the only eth account', () => {
```

Reply:
bit strange if there are non evm accounts/scopes but we can obviously modify this when that comes around.